driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/Fixture.java

```diff
@@ -74,49 +78,76 @@ public static MongoDatabase getDefaultDatabase() {
+    public static MongoCollection<Document> initializeCollection(final MongoNamespace namespace) {
         MongoDatabase database = getMongoClient().getDatabase(namespace.getDatabaseName());
         try {
-            ObservableSubscriber<Document> subscriber = new ObservableSubscriber<Document>();
-            database.runCommand(new Document("drop", namespace.getCollectionName())).subscribe(subscriber);
-            subscriber.await(10, SECONDS);
+            Mono.from(database.runCommand(new Document("drop", namespace.getCollectionName()))).block(TIMEOUT_DURATION);
         } catch (MongoCommandException e) {
             if (!e.getErrorMessage().contains("ns not found")) {
                 throw e;
             }
+        } catch (Throwable t) {
+            throw new RuntimeException(t);
         }
         return database.getCollection(namespace.getCollectionName());
     }
 
-    public static void dropDatabase(final String name) throws Throwable {
+    public static boolean isSharded() {
+        getMongoClient();
+        return mongoClient.getCluster().getDescription().getType() == SHARDED;
+    }
+
+    public static void dropDatabase(final String name) {
         if (name == null) {
             return;
         }
         try {
-            ObservableSubscriber<Document> subscriber = new ObservableSubscriber<Document>();
-            getMongoClient().getDatabase(name).runCommand(new Document("dropDatabase", 1)).subscribe(subscriber);
-            subscriber.await(10, SECONDS);
+            Mono.from(getMongoClient().getDatabase(name).runCommand(new Document("dropDatabase", 1))).block(TIMEOUT_DURATION);
         } catch (MongoCommandException e) {
             if (!e.getErrorMessage().contains("ns not found")) {
                 throw e;
             }
+        } catch (Throwable t) {
+            throw new RuntimeException(t);
         }
     }
 
-    public static void drop(final MongoNamespace namespace) throws Throwable {
+    public static void drop(final MongoNamespace namespace) {
         try {
-            ObservableSubscriber<Document> subscriber = new ObservableSubscriber<Document>();
-            getMongoClient().getDatabase(namespace.getDatabaseName())
-                    .runCommand(new Document("drop", namespace.getCollectionName()))
-                    .subscribe(subscriber);
-            subscriber.await(20, SECONDS);
+            Mono.from(getMongoClient().getDatabase(namespace.getDatabaseName())
+                              .runCommand(new Document("drop", namespace.getCollectionName()))).block(TIMEOUT_DURATION);
         } catch (MongoCommandException e) {
             if (!e.getErrorMessage().contains("ns not found")) {
                 throw e;
             }
+        } catch (Throwable t) {
+            throw new RuntimeException(t);
         }
     }
 
+    public static synchronized void waitForLastServerSessionPoolRelease() {
```

Comment:
What prompted the need for this?

Reply:
This was added as it existed in async and I didn't want it to be removed.